tests/test_freezing/test_freezing_compare.py

```diff
@@ -0,0 +1,187 @@
+        input_amendments_df = pd.DataFrame(data=data, columns=input_cols)
+
+        numeric_cols = [
+            "203", "204", "205", "206", "207", "209", "210",
+            "211", "212", "214", "216", "218", "219", "220", "221", "222",
+            "223", "225", "226", "227", "228", "229", "237", "242", "243",
+            "244", "245", "246", "247", "248", "249", "250", "405", "406",
+            "407", "408", "409", "410", "411", "412", "501", "502", "503",
+            "504", "505", "506", "507", "508", "602", "701", "702", "703",
+            "704", "705", "706", "707", "709", "711",
+        ]
+
+        for col in numeric_cols:
+            input_amendments_df[col] = None
+
+        return input_amendments_df
+
+
+    # Create expected outcome df
+    def create_test_expected_outcome_df(self) -> pd.DataFrame:
+        """Create a test expected_outcome df."""
+        input_cols = ["reference", "period", "instance", "201", "202", "200", "601", "201_diff", "202_diff", "200_diff", "601_diff", "accept_changes"]
+        data = [
+            ["A", 202412, 2.0, 1.0, 2.0, "A", None, 0.0, 0.0, None, None, False],
+            ["B", 202412, None, None, 1.0, "A", "B", None, 0.0, "A", "B", False],
+            ["C", 202412, 0.0, 2.0, 2.0, "A", "B", 1.0, 0.0, None, None, False],
+            ["D", 202412, 1.0, 2.0, 3.0, "E", "D", 0.0, 0.0, "E", None, False],
+            ["E", 202412, None, 10.0, 1.0, "E", "F", 6.0, -4.0, None, None, False],
+        ]
+        input_expected_outcome_df = pd.DataFrame(data=data, columns=input_cols)
+
+        numeric_cols = [
+            "203", "204", "205", "206", "207", "209", "210",
+            "211", "212", "214", "216", "218", "219", "220", "221", "222",
+            "223", "225", "226", "227", "228", "229", "237", "242", "243",
+            "244", "245", "246", "247", "248", "249", "250", "405", "406",
+            "407", "408", "409", "410", "411", "412", "501", "502", "503",
+            "504", "505", "506", "507", "508", "602", "701", "702", "703",
+            "704", "705", "706", "707", "709", "711",
+        ]
+
+        for col in numeric_cols:
+            input_expected_outcome_df[col] = None
+        for col in numeric_cols:
+            input_expected_outcome_df[f"{col}_diff"] = None
+
+        return input_expected_outcome_df
+
+
+    def test_get_amendments(self):
+        """Test for get_amendments()."""
+        # Create test dataframes
+        input_frozen_df = self.create_test_frozen_df()
+        input_amendments_df = self.create_test_amendments_df()
+        expected_outcome_df = self.create_test_expected_outcome_df()
+
+        # Run the function
+        result = get_amendments(
+            input_frozen_df, input_amendments_df
+        )
+
+        expected_outcome_df = expected_outcome_df[result.columns]
+
+        print(result)
+        # Check the output
+        assert_frame_equal(
+            expected_outcome_df, result
+        ), "get amends output is not as expected"
```

Comment:
It would be better to use get_amendments() in this message. Also picky but '.' at the end :)